### PR TITLE
[Az.DeviceRegistry] - Integrate Azure Device Registry APIs into Azure Powershell Commandlets for version 2024-11-01

### PR DESCRIPTION
**Pull request by mryanlo**
## Description

This PR integrates the Azure Device Registry APIs into the Azure Powershell Commandlets. I followed the instructions from the autorest autogenerating pwsh guide [here](https://eng.ms/docs/cloud-ai-platform/azure-core/azure-experiences-and-ecosystems/azure-portal-and-client-tools-ruhim/azure-cli-tools-azure-cli-powershell-and-terraform/azure-cli-tools/onboarding/azurepowershell/dev_guidance_codegen). This exposes the GA API version 2024-11-01 of ADRs ARM specs to make them available to call via powershell commandlets.

## Mandatory Checklist

- Please choose the target release of Azure PowerShell. (⚠️**Target release** is a different concept from **API readiness**. Please click below links for details.)
  - [x] [General release](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] [Public preview](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] [Private preview](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] [Engineering build](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] No need for a release

- [x] Check this box to confirm: **I have read the [_Submitting Changes_](../blob/main/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/main/CONTRIBUTING.md) and reviewed the following information:**

* **SHOULD** update `ChangeLog.md` file(s) appropriately
    * For SDK-based development mode, update `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`.
        * A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header in the past tense. 
    * For autorest-based development mode, include the changelog in the PR description.
    * Should **not** change `ChangeLog.md` if no new release is required, such as fixing test case only.
* **SHOULD** regenerate markdown help files if there is cmdlet API change. [Instruction](../blob/main/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)
* **SHOULD** have proper test coverage for changes in pull request.
* **SHOULD NOT** adjust version of module manually in pull request

### CODE REVIEW

File: src/DeviceRegistry/DeviceRegistry.Autorest/docs/Get-AzDeviceRegistryOperationStatus.md [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+
+# Get-AzDeviceRegistryOperationStatus
+
+## SYNOPSIS
+Returns the current status of an async operation.
+
+## SYNTAX
+
+```
+Get-AzDeviceRegistryOperationStatus -Location <String> -OperationId <String> [-SubscriptionId <String[]>]
```
Comment (mryanlo) [inline src/DeviceRegistry/DeviceRegistry.Autorest/docs/Get-AzDeviceRegistryOperationStatus.md:16]: Do we need this? Current pwsh commandlets for Create/Update resources automatically wait for async operation status to complete, as well as this pwsh command cannot get the async op uri signing params (see [here](https://armwiki.azurewebsites.net/api_contracts/AsyncOperationSigningAndValidation.html#async-operation-uri-signing) to successfully run

File: src/DeviceRegistry/DeviceRegistry.Autorest/README.md [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+  # Following two directives are v4 specific
+  - where:
+      variant: ^(Create|Update)(?!.*?(Expanded|JsonFilePath|JsonString))
+    remove: true
+  - where:
+      variant: ^CreateViaIdentity.*$
+    remove: true
+  # Remove GetViaIdentity and GetViaIdentityLocation from the Get-AzDeviceRegistryOperationStatus
+  - where:
+      variant: ^GetViaIdentity$|GetViaIdentityLocation$
```
Comment (VeryEarly) [inline src/DeviceRegistry/DeviceRegistry.Autorest/README.md:66]: remove this line so all operationsstatus related cmdlets will be removed

Comment (mryanlo) [reply src/DeviceRegistry/DeviceRegistry.Autorest/README.md]: done

**Review by VeryEarly — APPROVED** [review]

**Pull request merged**